docs/chain/did.md

```diff
@@ -0,0 +1,132 @@
+Beware that this also means that the DID creator gives up some power over the DID: The submitter who pays the deposit will be able to delete the DID from the blockchain and claim back its deposit.
+Once the `did::create` extrinsic is submitted and executed, the DID is written to the chain.
+
+## Using a DID
+
+Once the DID is successfully registered on chain, it can be used to perform certain on-chain actions that are not possible to do with a regular account. 
+This includes handling of attestations and ctypes, setting up trust hierarchies through delegations, managing web3 names and much more.
+
+Those actions need to be signed by the DID before they can be submitted to chain by any account that the DID owner specifies when signing.
+We are naming those actions "DID-Calls". To submit those there is a special extrinsic called `submit_did_call`. 
+
+The process of doing any DID-Call is always the same:
+
+* Construct the actual call you want to execute including all arguments of that extrinsic.
+* Wrap the call in a `DidAuthorizedCallOperation` together with
+    * Senders DID to indicate who wants this operation to happen
+    * Senders DID tx_counter + 1 to prevent replay attacks
+    * Current block number to prevent the operation to be submitted too far in the future
+    * Account of the submitter to allow the DID owner to specify who is allowed to submit
+* Create a signature over the `DidAuthorizedCallOperation` by scale-encoding it and signing it using the appropriate key
+    * Most operations require the authentication key of the DID to be used
+    * Managing attestations requires the attestation key
+    * Managing delegations requires the delegation key
+* Construct the `submit_did_call` extrinsic consisting of
+    * The `DidAuthorizedCallOperation`
+    * The DID signature
+* Pass the call over to the submitter who can now sign and submit it to the chain
+    * The submitter will have to pay for all fees and deposits that result from the operation
+    * In general the submitter will have the power to delete all on-chain objects to reclaim its deposit
+* The chain now checks that
+    * The submitters signature is correct
+    * The submitter is the one specified in the `DidAuthorizedCallOperation`
+    * The DID signature is correct
+    * The tx_counter is valid (current tx_counter + 1)
+    * The blocknumber is not older than an hour (given 12s block time)
+* After that the actual call gets dispatched with a special `DidOrigin`
+    * This allows the executer of the actual call to get the DID and the account of the submitter
+
+
+## Updating a DID
+
+There is a set of extrinsics available to update a full DID.
+These are:
+
+* `set_authentication_key`
+* `set_delegation_key`
+* `remove_delegation_key`
+* `set_attestation_key`
+* `remove_attestation_key`
+* `add_key_agreement_key`
+* `remove_key_agreement_key`
+* `add_service_endpoint`
+* `remove_service_endpoint`
+* `delete`
+
+
+<<<<<<< HEAD
```

Comment:
after the merge conflict is solved this can get merged. Sorry it pushed a commit in there that probably resulted in a few conflicts. thought it would be easier to show what I meant than explaining it... 😓

Reply:
Don't know how I could have missed that. Fixed now :man_shrugging: